src/SharedPipelineComponents/Datalock/SFA.DAS.CollectionEarnings.DataLock/Services/DatalockValidationService.cs

```diff
@@ -0,0 +1,295 @@
+            {
+                errors.Add(DataLockErrorCodes.NotLevyPayer);
+            }
+
+            if (commitments.Count > 1 && paymentType == TransactionTypesFlag.AllLearning)
+            {
+                errors.Add(DataLockErrorCodes.MultipleMatches);
+            }
+            else if (commitments.Count == 0)
+            {
+                CheckForEarlierStartDate(earning, allCommitments, censusDate, paymentType, result);
+                return;
+            }
+
+            result.AddDistinctRecords(earning, errors, paymentType, commitments.First());
+        }
+
+        private void CheckForEarlierStartDate(RawEarning earning, List<Commitment> commitments, DateTime censusDate, TransactionTypesFlag paymentType, DatalockValidationResult result)
+        {
+            var matchResult = _datalockMatcher.Match(commitments, earning, censusDate);
+            if (!matchResult.ErrorCodes.Any() && matchResult.Commitments.Any(x => x.Ukprn == earning.Ukprn))
+            {
+                if (commitments.Any(x => x.PausedOnDate.HasValue || x.PaymentStatus == 2))
+                {
+                    result.AddDistinctRecords(earning, new List<string> { DataLockErrorCodes.EmployerPaused }, paymentType,
+                        matchResult.Commitments.First());
+                }
+            }
+            else 
+            {
+                result.AddDistinctRecords(earning, matchResult.ErrorCodes, paymentType, matchResult.Commitments.Last());
+            }
+        }
+
+        private DateTime CalculateOnProgCensusDate(RawEarning earning)
+        {
+            var month = MonthFromPeriod(earning.Period);
+            var year = YearFromPeriod(earning.Period, earning.EpisodeStartDate ?? new DateTime(9999, 01, 01));
+
+            var lastDayOfMonth = DateTime.DaysInMonth(year, month);
+            return new DateTime(year, month, lastDayOfMonth);
+        }
+
+        private int MonthFromPeriod(int period)
+        {
+            if (period < 6)
+            {
+                return period + 7;
+            }
+
+            return period - 5;
+        }
+
+        private int YearFromPeriod(int period, DateTime episodeStartDate)
+        {
+            var month = MonthFromPeriod(period);
+            var startOfAcademicYear = StartOfAcademicYearFromEpisodeStartDate(episodeStartDate);
+            if (month < 8)
+            {
+                return startOfAcademicYear.Year + 1;
+            }
+            return startOfAcademicYear.Year;
+        }
+
+        private DateTime StartOfAcademicYearFromEpisodeStartDate(DateTime episodeStartDate)
+        {
+            var month = episodeStartDate.Month;
+            if (month < 8)
+            {
+                return new DateTime(episodeStartDate.Year - 1, 8, 1);
+            }
+            return new DateTime(episodeStartDate.Year, 8, 1);
+        }
+    }
+
+    public class DatalockValidationResult
```

Comment:
A quick question of styles. If a class is public do we not have it in it's own file. The same with Interfaces etc?